### PR TITLE
New package: randomx-1.1.8

### PR DESCRIPTION
**Pull request by kevcrumb**
~*randomx* contains binaries for benchmarking a machine for RandomX algo~
~*librandomx* provides library for using the algo (for example from a mining software)~

*randomx* provides library for using the algo (for example from a mining software)

### CODE REVIEW

**Comment by kevcrumb** [comment]
Couldn't get arm6 to build, but pkg should be good for inclusion.

File: srcpkgs/randomx/template [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+case "$XBPS_TARGET_MACHINE" in
+	armv7*) configure_args+=" -DARCH=armv7" ;;
+	armv6*) configure_args+=" -DARCH=armv6" ;;
+	armv5*) configure_args+=" -DARCH=armv5" ;;
+	aarch64*) configure_args+=" -DARCH=aarch64" ;;
+	mips*) configure_args+=" -DARCH=mipsr32 -DNO_AES=ON" ;;
+	ppc64le*) configure_args+=" -DARCH=ppc64le" ;;
+	ppc64*) configure_args+=" -DARCH=ppc64" ;;
+	ppc*) configure_args+=" -DARCH=ppc"     ;;
+#        *) configure_args+=" -DARCH=native"     ;;
```
Comment (q66) [inline srcpkgs/randomx/template:28]: replace this with

```
	*) broken="unsupported architecture";;
```

**Comment by sgn** [comment]
On 2020-11-25 07:57:25-0800, Daniel Kolesa <notifications@github.com> wrote:
> @q66 commented on this pull request.
> 
> 
> 
> > +
> +if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
> +	makedepends+=" libatomic-devel"
> +	configure_args+=" -DCMAKE_CXX_STANDARD_LIBRARIES=-latomic"
> +fi
> +
> +case "$XBPS_TARGET_MACHINE" in
> +	armv7*) configure_args+=" -DARCH=armv7" ;;
> +	armv6*) configure_args+=" -DARCH=armv6" ;;
> +	armv5*) configure_args+=" -DARCH=armv5" ;;
> +	aarch64*) configure_args+=" -DARCH=aarch64" ;;
> +	mips*) configure_args+=" -DARCH=mipsr32 -DNO_AES=ON" ;;
> +	ppc64le*) configure_args+=" -DARCH=ppc64le" ;;
> +	ppc64*) configure_args+=" -DARCH=ppc64" ;;
> +	ppc*) configure_args+=" -DARCH=ppc"     ;;
> +#        *) configure_args+=" -DARCH=native"     ;;
> 
> replace this with
> 
> ```
> 	*) broken="unsupported architecture" ;;
> ```

Doesn't "*" include i686 and x86_64, which should be always supported?

**Comment by q66** [comment]
oh yeah

you'll need to fill in dedicated lines for those

**Comment by kevcrumb** [comment]
> also why did you remove the ppc ones?

Had read your instruction like that.

I see Monero just sets `default` (https://github.com/void-linux/void-packages/blob/master/srcpkgs/monero/template#L37).

Pushed with explicit `i686` & `x86_64` now.

File: srcpkgs/randomx/template [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	armv7*) configure_args+=" -DARCH=armv7" ;;
+	armv6*) configure_args+=" -DARCH=armv6" ;;
+	armv5*) configure_args+=" -DARCH=armv5" ;;
+	aarch64*) configure_args+=" -DARCH=aarch64" ;;
+        i686*) configure_args+=" -DARCH=i686" ;;
+	mips*) configure_args+=" -DARCH=mipsr32 -DNO_AES=ON" ;;
+	ppc64le*) configure_args+=" -DARCH=ppc64le" ;;
+	ppc64*) configure_args+=" -DARCH=ppc64" ;;
+	ppc*) configure_args+=" -DARCH=ppc" ;;
+        x86_64*) configure_args+=" -DARCH=x86_64" ;;
```
Comment (ericonr) [inline srcpkgs/randomx/template:29]: fix indentation

**Comment by q66** [comment]
> Had read your instruction like that.

not really, i commented on just one line

what monero sets does not matter, it's wrong there too, since packages like to assume `x86_64` and `i686` won't be cross-compiled, we're fixing that

**Pull request closed without merging**